api-gateway/src/main/kotlin/com/saveourtool/save/gateway/controller/SecurityInfoController.kt

```diff
@@ -30,9 +35,17 @@ class SecurityInfoController(
     /**
      * Endpoint that provides the information about the current logged-in user (powered by spring security and OAUTH)
      *
-     * @param principal
+     * @param authentication
      * @return user information
      */
     @GetMapping("/user")
-    fun currentUserName(principal: Principal?): String? = principal?.userName()
+    fun currentUserName(authentication: Authentication?): Mono<String> = when (authentication) {
+        is UsernamePasswordAuthenticationToken -> authentication.name.toMono()
+        is OAuth2AuthenticationToken -> {
+            val source = authentication.authorizedClientRegistrationId
+            val nameInSource = authentication.name
+            backendService.findByOriginalLogin(source, nameInSource).map { it.username }
```

Comment:
This thing is still worrying me: what will be in original login? user-name-attribute?
Or our own ID? If our own ID - why do we need source?

Reply:
some id from source provider: user id for github and bunch of numbers for google